### PR TITLE
kubeadm: revisit the "Network setup" section post-feedback

### PR DESCRIPTION
**Pull request by neolit123**
- In a early note explain that there could be multiple default gateways on a host.
- Do not have 1 note, 1 warning, then other note.
- Leave the warning that the "custom IP" setup is not recommended at the end of the section.

follow up on https://github.com/kubernetes/website/pull/43872#pullrequestreview-1744902982

### CODE REVIEW

**Comment by neolit123** [comment]
/sig cluster-lifecycle
/assign @sftim 
PTAL for LGTM/approve

File: content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md [hunk 1]
```diff
@@ -89,7 +89,7 @@ After you initialize your control-plane, the kubelet runs normally.
 #### Network setup
 
 kubeadm similarly to other Kubernetes components tries to find a usable IP on
-the network interface associated with the default gateway on a host. Such
+the network interfaces associated with a default gateway on a host. Such
```
Comment (neolit123) [inline content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md:92]: pluralize

File: content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md [hunk 2]
```diff
@@ -98,10 +98,22 @@ To find out what this IP is on a Linux host you can use:
...
+{{< /note >}}
+
 Kubernetes components do not accept custom network interface as an option,
 therefore a custom IP address must be passed as a flag to all components instances
 that need such a custom configuration.
 
+{{< note >}}
+If the host does not have a default gateway and if a custom IP address is not passed
+to a Kubernetes component, the component may exit with an error.
+{{< /note >}}
```
Comment (neolit123) [inline content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md:115]: could be outside of a note, but better to have it more visible in the render

**Review by sftim — COMMENTED** [review]
Thanks

/lgtm

**Review by nate-double-u — COMMENTED** [review]
Good updates, thanks @neolit123 
/approve

**Pull request merged**